src/client.cpp

```diff
@@ -127,11 +128,18 @@ RawVector fetch(CharacterVector server, CharacterVector method, RawVector reques
   
   RGRPC_LOG("Starting batch...");
   error = grpc_call_start_batch(c, ops, (size_t)(op - ops), tag(1), NULL);
-  grpc_completion_queue_next(cq, deadline, RESERVED); //actually does the work
-  
-  
-  
-  
+  if (error != GRPC_CALL_OK) {
```

Comment:
No idea why, but these still do not capture the timeouts / connection refused errors :/
@nfultz, any hints on this?

Reply:
hmmm, odd.

Reply:
If you have some time, please give it a try:

```r
library(grpc)
spec <- system.file('examples/helloworld.proto', package = 'grpc')
impl <- read_services(spec)
client <- grpc_client(impl, "foo.bar:50051")
who <- client$SayHello$build(name = 'who')
client$SayHello$call(who)
```

Reply:
The client-side segfaults are also fixed at https://github.com/daroczig/grpc/pull/3